### PR TITLE
Fix submodules by calling update init

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Small fix for ci darwin. The submodules were not being pulled in automatically, this pr adds a line to initialize the submodules before calling cmake. 

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented. NA
- [ ] Adds a test for any bugs fixed. Adds tests for new features. NA
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Comment by JoshuaSBrown** [comment]
Link to working ci https://re-git.lanl.gov/eap-oss/parthenon/-/jobs/90384

File: scripts/darwin/build_fast.sh [hunk 1]
```diff
@@ -8,6 +8,9 @@ source /projects/parthenon-int/parthenon-project/.bashrc
 # Calculate number of available cores
 export J=$(( $(nproc --all) )) && echo Using ${J} cores during build
 
+# Ensure the submodules are pulled in
+git submodule update --init
```
Comment (Yurlungur) [inline scripts/darwin/build_fast.sh]:
```suggestion
git submodule update --init --recursive
```

**Review by Yurlungur — APPROVED** [review]
This works. Minor suggestion below. Also, why not use the `submodule strategy recursive` setting in `gitlab`?
https://docs.gitlab.com/ee/ci/git_submodules.html

**Comment by pgrete** [comment]
Or directly using the GitLab feature: https://github.com/lanl/parthenon/blob/9cb7c380fba4381f035ec8b89d1953ae0ac16878/.gitlab-ci.yml#L20

**Review by AndrewGaspar — APPROVED** [review]

**Comment by JoshuaSBrown** [comment]
@pgrete please approve this.

**Review by pgrete — APPROVED** [review]
Here we go.
(Sorry for the delay. I didn't think my approval was needed here as this is a "trivial" change and for the LANL infrastructure).

**Comment by JoshuaSBrown** [comment]
> Here we go.
> (Sorry for the delay. I didn't think my approval was needed here as this is a "trivial" change and for the LANL infrastructure).

Maybe it wasn't but the merge was blocked, might have been because it was out of date with develop. Thanks.

**Comment by AndrewGaspar** [comment]
I'm going to merge this - not worried about this effecting the existing CI.

**Pull request merged**